theano/tensor/tests/test_slinalg.py

```diff
@@ -229,6 +231,27 @@ def test_solve_correctness(self):
         assert numpy.allclose(scipy.linalg.solve_triangular(U_val, b_val, lower=False),
                               upper_solve_func(U_val, b_val))
 
+    def test_solve_dtype(self):
+        if not imported_scipy:
+            raise SkipTest("Scipy needed for the Solve op.")
+
+        dtypes = ['int8', 'int16', 'int32', 'int64',
+                  'float32', 'float64']
+
+        A_val = numpy.eye(2)
+        b_val = numpy.ones((2, 1))
+
+        for A_dtype, b_dtype in itertools.product(dtypes, dtypes):
+            A = tensor.matrix(dtype=A_dtype)
+            b = tensor.matrix(dtype=b_dtype)
+            x = solve(A, b)
+            fn = function([A, b], x)
+            x_result = fn(A_val.astype(A_dtype), b_val.astype(b_dtype))
+            print(A_dtype, b_dtype)
+            print('x', x.dtype, x_result.dtype, x.dtype == x_result.dtype)
```

Comment:
We don't need the print() in the final code, nose_tests will print the error message from the assert if it fails.